### PR TITLE
feat(addon): #775 Remove ContentSearch dependency

### LINKED ISSUE #762: SearchProvider.jsm collects suggestions for search box
The search provider as of right now only does the bare minimum of getting the initial state of search and performs search. It needs to be able to get suggestions for search too

### PR DESCRIPTION
**Pull request by sarracini**
@rlr this is the addon-side work.
This closes #762 #763 #764 #766 #775 

@oyiptong if you want to review also feel free

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/activity-stream/797)

### CODE REVIEW

**Comment by sarracini** [comment]
Just as a note, this needs to land TOGETHER with  #798, or else nothing will work. Hence the travis failures

File: lib/ActivityStreams.js [hunk 1]
```diff
@@ -185,25 +185,53 @@ ActivityStreams.prototype = {
     this.send(am.actions.Response(responseType, cachedLinks, {append}), worker);
   },
 
+  /**
+   * Responds to search requests
+   */
   _respondToSearchRequests({msg, worker}) {
+    const win = windowMediator.getMostRecentWindow("navigator:browser");
+    const gBrowser = win.getBrowser();
+    const browser = gBrowser.selectedBrowser;
```
Comment (rlr) [inline lib/ActivityStreams.js]: There is a browser inside a browser inside a window. #mindblown

File: lib/ActivityStreams.js [hunk 2]
```diff
@@ -185,25 +185,53 @@ ActivityStreams.prototype = {
...
+      case am.type("NOTIFY_REMOVE_FORM_HISTORY_ENTRY"): {
+        let entry = msg.data;
+        SearchProvider.search.removeFormHistoryEntry(browser, entry);
+        break;
+      }
+      case am.type("NOTIFY_MANAGE_ENGINES"):
+        SearchProvider.search.manageEngines(browser);
+        break;
+      case am.type("SEARCH_CYCLE_CURRENT_ENGINE_REQUEST"):
+        SearchProvider.search.asyncCycleEngine(msg.data);
```
Comment (rlr) [inline lib/ActivityStreams.js]: I think you need a `yield` or `.then(....` here, right?

File: lib/SearchProvider.js [hunk 3]
```diff
@@ -46,35 +65,210 @@ NewTabSearchProvider.prototype = {
...
+        iconBuffer: favicon,
+      });
+    }
+    return Object.assign({}, state);
+  }),
+
+  /**
+   *  Gets the current engine - a combination of the engine name and the icon URI.
+   */
+  asyncGetCurrentEngine: Task.async(function*() {
```
Comment (rlr) [inline lib/SearchProvider.js]: I don't see why this one needs to be async. Everything below looks sync.

**Comment by rlr** [comment]
I think there are a few async related issues but otherwise looks pretty good. I'll go through the content side next. That looks like a big one.

**Pull request closed without merging**